components/script/dom/readablestream.rs

```diff
@@ -390,10 +423,13 @@ pub enum ExternalUnderlyingSource {
 }
 
 #[derive(JSTraceable, MallocSizeOf)]
+#[unrooted_must_root_lint::must_root]
 struct ExternalUnderlyingSourceController {
     /// Loosely matches the underlying queue,
     /// <https://streams.spec.whatwg.org/#internal-queues>
     buffer: RefCell<Vec<u8>>,
+    #[ignore_malloc_size_of = "Rc is hard"]
+    pending_reader: RefCell<Option<Rc<Promise>>>,
```

Comment:
I would rather keep those on the dom struct, in some ways `ExternalUnderlyingSourceController` doesn't really belong here at all because it's the "native" pure Rust part, and should eventually perhaps be separated out into `shared/script`(similar to Blob and Messageport).